engine/src/w32-ds-player.cpp

```diff
@@ -676,6 +735,71 @@ bool MCWin32DSPlayer::SetVisible(bool isVisible)
 	return S_OK == pVideo->put_Visible(isVisible ? OATRUE : OAFALSE );
 }
 
+static inline bool _mediasubtype_supports_mirror(const GUID &p_subtype)
+{
+	return p_subtype == MEDIASUBTYPE_ARGB32;
+}
+
+// check the connection video subformat and attempt to reconnect if it'll break video mirroring.
+static bool MCWin32DSReconnectVideoRendererConnection(IGraphBuilder *p_graph, IBaseFilter *p_vmr9)
+{
+	bool t_success = true;
+
+	CComPtr<IPin> t_pin;
+	if (t_success)
+		t_success = MCWin32DSFilterGetFirstPin(p_vmr9, PINDIR_INPUT, t_pin);
+
+	AM_MEDIA_TYPE t_current_mediatype;
+	MCMemoryClear(t_current_mediatype);
+	if (t_success)
+		t_success = SUCCEEDED(t_pin->ConnectionMediaType(&t_current_mediatype));
+
+	AM_MEDIA_TYPE t_new_mediatype;
+	MCMemoryClear(t_new_mediatype);
+
+	// If the current connection media type is not compatible then look for one that is
+	if (t_success && !_mediasubtype_supports_mirror(t_current_mediatype.subtype))
+	{
+		CComPtr<IPin> t_other_pin;
+		if (t_success)
+			t_success = SUCCEEDED(t_pin->ConnectedTo(&t_other_pin));
+
+		CComPtr<IEnumMediaTypes> t_mediatype_enum;
+		if (t_success)
+			t_success = SUCCEEDED(t_other_pin->EnumMediaTypes(&t_mediatype_enum));
+
+		AM_MEDIA_TYPE *t_mt = nil;
+		ULONG t_mt_fetched = 0;
+		while (t_success && (t_new_mediatype.majortype == MEDIATYPE_NULL) && (S_OK == t_mediatype_enum->Next(1, &t_mt, &t_mt_fetched)))
```

Comment:
@livecodeian probably not worth changing to clarify it as it's clear in the docs but you had me wondering for a bit if `t_mt_fetched` needed checking but it seems this param is ignored/can be null if `cMediaTypes` is 1.